server.py

```diff
@@ -29,6 +30,26 @@ def list():
 			cache_timeout=0)
 
 
+@app.route("/geoip")
+def geoip():
+	ip = request.remote_addr
+
+	if ip.startswith("::ffff:"):
+		ip = ip[7:]
+	try:
+		geo = reader.get(ip)
+	except geoip2.errors.GeoIP2Error:
+		app.logger.warning("GeoIP lookup failure for %s." % (ip,))
+		return "unknown", 500
+
+	if geo and "continent" in geo:
+		return geo["continent"]["code"], 200
```

Comment:
Also, caching headers should be set for the request:
```python
	resp = flask.Response(the_content)
	resp.cache_control.max_age = 7*24*60*60
	resp.cache_control.public = True
	return resp
```

Reply:
good idea